src/Temporalio/Client/EnvConfig/ClientEnvConfig.cs

```diff
@@ -0,0 +1,414 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Temporalio.Client.EnvConfig
+{
+    /// <summary>
+    /// Represents the overall client configuration.
+    /// </summary>
+    /// <param name="Profiles">The configuration profiles.</param>
+    public sealed record ClientEnvConfig(IReadOnlyDictionary<string, ClientEnvConfig.Profile> Profiles)
+    {
+        /// <summary>
+        /// Load client configuration from environment variables and configuration files.
+        /// </summary>
+        /// <param name="options">Options for loading the configuration.</param>
+        /// <returns>Loaded configuration data.</returns>
+        public static ClientEnvConfig Load(ConfigLoadOptions options)
+        {
+            var runtime = Runtime.TemporalRuntime.Default.Runtime;
+            var profiles = Bridge.EnvConfig.LoadClientConfig(runtime, options);
+            return new ClientEnvConfig(profiles);
+        }
+
+        /// <summary>
+        /// Load client connection options directly from configuration.
+        /// </summary>
+        /// <param name="options">Options for loading the configuration profile.</param>
+        /// <returns>Client connection options.</returns>
+        public static TemporalClientConnectOptions LoadClientConnectOptions(ProfileLoadOptions options)
```

Comment:
```suggestion
        public static TemporalClientConnectOptions LoadClientConnectOptions(ProfileLoadOptions? options = null)
```

We should make the options parameter optional on all of these if the default `new()` is acceptable. Can do `options ??= new()` in the method just before use if needed to be non-null in code. But I think we want our samples code to just be able to `TemporalClient.ConnectAsync(ClientEnvConfig.LoadClientConnectOptions())`

Reply:
options are nullable and use fallback defaults